### ISSUE TITLE
v3.1.0 and openrtb.proto

### ISSUE THREAD

**Issue by va5ja** [google/protobuf#2179]
Hi there,

I'm getting some errors trying to generate PHP classes with `protoc --php_out=php openrtb.proto` using version 3.1.0 (default proto2 syntax):

```
openrtb.proto:2045:3: When enum name is stripped and label is PascalCased (Iab11), this value label conflicts with IAB1_1
openrtb.proto:2051:3: When enum name is stripped and label is PascalCased (Iab12), this value label conflicts with IAB1_2
openrtb.proto:2055:3: When enum name is stripped and label is PascalCased (Iab13), this value label conflicts with IAB1_3
openrtb.proto:2068:3: When enum name is stripped and label is PascalCased (Iab14), this value label conflicts with IAB1_4
openrtb.proto:2077:3: When enum name is stripped and label is PascalCased (Iab15), this value label conflicts with IAB1_5
openrtb.proto:2088:3: When enum name is stripped and label is PascalCased (Iab16), this value label conflicts with IAB1_6
openrtb.proto:2096:3: When enum name is stripped and label is PascalCased (Iab17), this value label conflicts with IAB1_7
openrtb.proto:2213:3: When enum name is stripped and label is PascalCased (Iab21), this value label conflicts with IAB2_1
openrtb.proto:2214:3: When enum name is stripped and label is PascalCased (Iab211), this value label conflicts with IAB2_11
openrtb.proto:2215:3: When enum name is stripped and label is PascalCased (Iab212), this value label conflicts with IAB2_12
openrtb.proto:2216:3: When enum name is stripped and label is PascalCased (Iab213), this value label conflicts with IAB2_13
openrtb.proto:2217:3: When enum name is stripped and label is PascalCased (Iab22), this value label conflicts with IAB2_2
openrtb.proto:2218:3: When enum name is stripped and label is PascalCased (Iab221), this value label conflicts with IAB2_21
openrtb.proto:2219:3: When enum name is stripped and label is PascalCased (Iab222), this value label conflicts with IAB2_22
openrtb.proto:2220:3: When enum name is stripped and label is PascalCased (Iab223), this value label conflicts with IAB2_23
openrtb.proto:2222:3: When enum name is stripped and label is PascalCased (Iab23), this value label conflicts with IAB2_3
openrtb.proto:2233:3: When enum name is stripped and label is PascalCased (Iab24), this value label conflicts with IAB2_4
openrtb.proto:2234:3: When enum name is stripped and label is PascalCased (Iab25), this value label conflicts with IAB2_5
openrtb.proto:2242:3: When enum name is stripped and label is PascalCased (Iab26), this value label conflicts with IAB2_6
```

I tried this using https://github.com/google/openrtb/blob/master/openrtb-core/src/main/protobuf/openrtb.proto and also https://developers.google.com/ad-exchange/rtb/downloads/openrtb-proto.txt but both files produce the same result. Is there any quick solution for this?

**Comment by TeBoring** [comment]
Just like the error suggested, these conflicting enum values names have the same pascal-cased name, which may cause conflicts in some code generators (This is not specific to php. All languages are affected). You can rename these conflicting names, e.g., IAB1_1 => IAB1_A1

**Comment by xfxyjwf** [comment]
The proto will no longer compile for any language when using 3.1.0 protocol compiler.

@haberman we probably should make the enum name check a warning instead of an error.

**Comment by haberman** [comment]
Makes sense. The warning should probably be that this proto will not be able to be compiled for some languages (like C#).

**Comment by acozzette** [comment]
@va5ja I just merged a fix to the 3.1.x branch and that should get merged to master soon. Let me know if you still run into any problems with that fix.
